quality_check/10-test_compendium_PLIER.Rmd

```diff
@@ -0,0 +1,308 @@
+ZFIN identifiers so that the IDs in the prior information and the expression
+data match.
+
+```{r}
+# first we need to read in the 
+exprs_file <- file.path("data", "larger_compendium_exprs_mat.RDS")
+exprs_mat <- readRDS(exprs_file)
+```
+
+We expect that there will be a 1:many mapping here, so we'll use the option
+where only the first thing that comes back is returned by setting 
+`multiVals = "first"` (note that this is the default behavior).
+
+```{r}
+# ENSDARG -> ZDB-GENE
+zfin_ids <- mapIds(org.Dr.eg.db, keys = rownames(exprs_mat),
+                   column = "ZFIN", keytype = "ENSEMBL",
+                   multiVals = "first")
+
+# these should be in the same order, but out of an abundance of caution!
+if (all.equal(rownames(exprs_mat), names(zfin_ids))) {
+  rownames(exprs_mat) <- zfin_ids
+} else {
+  stop("Something went wrong!")
+}
+```
+
+## PLIER
+
+We're going to do five repeats.
+
+```{r}
+set.seed(123)
+seeds <- sample(1:100000, 5)
+names(seeds) <- seeds
+```
+
+### Both technologies
+
+```{r}
+# find the common genes between the expression matrix and the prior information
+# matrix
+common_genes <- PLIER::commonRows(exprs_mat, combine_path_mat)
+
+# this expression matrix includes samples from both technologies
+both_tech_plier_list <- 
+  lapply(seeds,
+         function(x) PLIERWrapper(exprs_mat = exprs_mat,
+                                  common_genes = common_genes,
+                                  pathway_mat = combine_path_mat,
+                                  seed = x)
+  )
+```
+
+Check the pathway coverage for these models
+
+```{r}
+both_tech_pathway_coverage <- reshape2::melt(lapply(both_tech_plier_list, 
+                                                    GetPathwayCoverage))
+```
+
+### Microarray only
+
+```{r}
+microarray_mat <- exprs_mat[, -grep("SRR|ERR|DRR", colnames(exprs_mat))]
+microarray_plier_list <-
+  lapply(seeds, 
+         function(x) PLIERWrapper(exprs_mat = microarray_mat,
+                                  common_genes = common_genes,
+                                  pathway_mat = combine_path_mat,
+                                  seed = x)
+  )
+microarray_pathway_coverage <- reshape2::melt(lapply(microarray_plier_list,
+                                                     GetPathwayCoverage))
+```
+
+### RNA-seq only
+
+```{r}
+rnaseq_mat <- exprs_mat[, grep("SRR|ERR|DRR", colnames(exprs_mat))]
+rnaseq_plier_list <- 
+  lapply(seeds, 
+         function(x) PLIERWrapper(exprs_mat = rnaseq_mat,
+                                  common_genes = common_genes,
+                                  pathway_mat = combine_path_mat,
+                                  seed = x )
+         )
+rnaseq_pathway_coverage <- reshape2::melt(lapply(rnaseq_plier_list, 
+                                                 GetPathwayCoverage))
+```
+
+#### Wrangle pathway coverage results 
+
+And write to file
+
+```{r}
+coverage_df <- dplyr::bind_rows(list(
```

Comment:
Now that I'm seeing that you are combining these into rows, I think streamlining the above repeated chunks into a single chunk should be not too bad.